pegin/getaddress.go

```diff
@@ -0,0 +1,142 @@
+package pegin
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+
+	"github.com/btcsuite/btcutil"
+
+	"github.com/btcsuite/btcd/txscript"
+
+	"github.com/vulpemventures/go-elements/network"
+	"github.com/vulpemventures/go-elements/payment"
+
+	"github.com/btcsuite/btcd/chaincfg"
+
+	"github.com/vulpemventures/go-elements/address"
+
+	"github.com/btcsuite/btcd/btcec"
+)
+
+const (
+	MainNet NetworkType = iota
+	RegtestNet
+)
+
+type NetworkType int
+
+type AddressInfo struct {
+	ClaimScript      string
+	MainChainAddress string
+}
+
+// GetAddressInfo simulates getpeginaddress RPC call which returns mainChain
+// address to which BTC should be sent in BTC network and claim script that is
+// to be used in claimpegin RPC alongside with bitcoinTx and txoutproof
+func GetAddressInfo(
+	privateKey btcec.PrivateKey,
```

Comment:
better using a []byte and/or string/hex since is an high level method already.

Reply:
Done